python/smqtk/algorithms/descriptor_generator/caffe_descriptor.py

```diff
@@ -393,20 +393,20 @@ def check_get_uuid(d):
                     batch_uuids = \
                         uuid4proc[g*self.batch_size:(g+1)*self.batch_size]
                     self._process_batch(batch_uuids, data_elements,
-                                        descr_elements, procs)
+                                        descr_elements, procs, **kwds)
 
             if tail_size:
                 batch_uuids = uuid4proc[-tail_size:]
                 self._log.debug("Starting tail batch (size=%d)",
                                 len(batch_uuids))
                 self._process_batch(batch_uuids, data_elements, descr_elements,
-                                    procs)
+                                    procs, **kwds)
 
         self._log.debug("forming output dict")
         return dict((data_elements[k], descr_elements[k])
                     for k in data_elements)
 
-    def _process_batch(self, uuids4proc, data_elements, descr_elements, procs):
+    def _process_batch(self, uuids4proc, data_elements, descr_elements, procs, **kwds):
```

Comment:
I think we should be more explicit here and have a ``use_mp`` argument here instead of passing the full ``kwds`` map through.